harmonica/filters/_filters.py

```diff
@@ -62,3 +62,598 @@ def derivative_upward_kernel(fft_grid, order=1):
+    **downward** coordinates.
+    Each of these components can be obtained from the inclination and
+    declination angles of the geomagnetic field (:math:`I` and :math:`D`,
+    respectively) and for the magnetization vector (:math:`I_m` and
+    :math:`D_m`, respectively):
+
+    .. math::
+
+        \begin{cases}
+            f_e = \sin D \cos I \\
+            f_n = \cos D \cos I \\
+            f_u = \sin I
+        \end{cases}
+
+    .. math::
+
+        \begin{cases}
+            m_e = \sin D_m \cos I_m \\
+            m_n = \cos D_m \cos I_m \\
+            m_u = \sin I_m
+        \end{cases}
+
+    Parameters
+    ----------
+    fft_grid : :class:`xarray.DataArray`
+        Array with the Fourier transform of the original grid.
+        Its dimensions should be in the following order:
+        *freq_northing*, *freq_easting*.
+        Use :func:`xrft.xrft.fft` and :func:`xrft.xrft.ifft` functions to
+        compute the Fourier Transform and its inverse, respectively.
+    inclination : float in degrees
+        The inclination inducing Geomagnetic field.
+    declination : float in degrees
+        The declination inducing Geomagnetic field.
+    magnetization_inclination : float in degrees
+        The inclination of the total magnetization of the anomaly source.
+        Default is i, neglecting remanent magnetization and
+        self demagnetization.
+    magnetization_declination : float in degrees
+        The declination of the total magnetization of the anomaly source.
+        Default is d, neglecting remanent magnetization and
+        self demagnetization.
+
+    Returns
+    -------
+    da_filter : :class:`xarray.DataArray`
+        Array with the kernel for the reduction to the pole filter in frequency
+        domain.
+
+    References
+    ----------
+    [Blakely1995]_
+
+    See also
+    --------
+    harmonica.reduction_to_pole
+    """
+    # Check if magnetization angles are valid
+    _check_magnetization_angles(magnetization_inclination, magnetization_declination)
+    # Define magnetization angles if they are None
+    if magnetization_declination is None and magnetization_inclination is None:
+        magnetization_inclination = inclination
+        magnetization_declination = declination
+    # Catch the dims of the Fourier transformed grid
+    dims = fft_grid.dims
+    # Grab the coordinates of the Fourier transformed grid
+    freq_easting = fft_grid.coords[dims[1]]
+    freq_northing = fft_grid.coords[dims[0]]
+    # Convert frequencies to wavenumbers
+    k_easting = 2 * np.pi * freq_easting
+    k_northing = 2 * np.pi * freq_northing
+    # Compute the filter for reduction to pole in frequency domain
+    da_filter = _get_rtp_filter(
+        k_easting,
+        k_northing,
+        inclination,
+        declination,
+        magnetization_inclination,
+        magnetization_declination,
+    )
+    # Deal with inf and nan value
+    da_filter.data = np.nan_to_num(da_filter.data, posinf=0, nan=0)
```

Comment:
Why do we need this line? Where can we find `nan`s in the filter? And why should we set them to 0 (meaning that we turning that frequency off in the output)? Is it possible to catch any potential singularity before an not recurring to overwriting the `nan`s?

Reply:
That's a lazy implication that replaces the 0 wavenumbers with 0. I change it to `da_filter.loc[dict(freq_northing=0, freq_easting=0)] = 0`

Reply:
I see. I think explicitly changing the value of the filter for the zero wavenumber vector is more explicit and less prone to errors. It looks much better now!